scipy/interpolate/_aaa.py

```diff
@@ -309,6 +319,72 @@ def _compute_weights(self, f, z, rtol, max_terms):
         self.support_values = fj[i_non_zero]
         self.weights = wj[i_non_zero]
         self.errors = errors[: m + 1]
+        self._points = z
+        self._values = f
+
+    def clean_up(self, cleanup_tol=1e-13):
+        """Automatic removal of Froissart doublets.
+
+        Parameters
+        ----------
+        cleanup_tol : float, optional
+            Poles with residues less than this number times the geometric mean
+            of `values` times the minimum distance to `points` are deemed spurious by
+            the cleanup procedure, defaults to 1e-13.
+        """
+        # Find negligible residues
+        if any(self._values):
+            geom_mean_abs_f = scipy.stats.gmean(
+                np.abs(self._values[np.nonzero(self._values)])
+            )
+        else:
+            geom_mean_abs_f = 0
+
+        Z_distances = np.min(
+            np.abs(np.subtract.outer(self.poles(), self._points)), axis=1
+        )
+
+        with np.errstate(divide="ignore", invalid="ignore"):
+            ii = np.flatnonzero(
+                np.abs(self.residues()) / Z_distances < cleanup_tol * geom_mean_abs_f
+            )
+
+        ni = ii.size
+        if ni == 0:
+            return
+
+        warnings.warn(f"{ni} Froissart doublets detected.", RuntimeWarning,
+                        stacklevel=2)
```

Comment:
Too bad python warnings are only issued only once. Maybe make this function return `ni` or, better, the doublets themselves.

Reply:
This seems to work fine for me?
```
In [1]: import numpy as np

In [2]: from scipy.interpolate import AAA

In [3]: z = np.exp(np.linspace(-0.5, 0.5 + 15j*np.pi, num=1000))

In [4]: f = lambda z: np.log(2+z**4)/(1-16*z**4)

In [8]: r = AAA(z, f(z), rtol=0, clean_up=False)
/home/jakeb/development/scipy/build-install/lib/python3.11/site-packages/scipy/interpolate/_aaa.py:224: RuntimeWarning: AAA failed to converge within 100 iterations.
  self._compute_weights(z, f, rtol, max_terms)

In [10]: r.clean_up(cleanup_tol=1e-9)
<ipython-input-10-fcb96bec813b>:1: RuntimeWarning: 3 Froissart doublets detected.
  r.clean_up(cleanup_tol=1e-9)

In [11]: r.clean_up(cleanup_tol=1e-5)
<ipython-input-11-60d1c1a70f72>:1: RuntimeWarning: 1 Froissart doublets detected.
  r.clean_up(cleanup_tol=1e-5)
```

Reply:
Something messed up with warnings filters, maybe `ipython` itself (https://github.com/ipython/ipython/issues/11207

Reply:
Hmm ok, return the doublets themselves it is not possible though

Reply:
> return the doublets themselves it is not possible though

Why not? 
https://github.com/scipy/scipy/pull/21373/files#diff-0f5a7ffb404523925b3efec69fced6d25c64be10d9b22ecfe9481e12e02b60a6R359 and below

Reply:
You could return the removed support points but not the doublets

Reply:
My only point is that python warnings are not a reliable mechanism to signal something to users. Any return value would do here IMO. Even the difference in lengths of the support before and after cleanup.

Reply:
Ah right sorry took me a while to understand the point here sorry!

The method now returns the number of detected Froissart doublets.